Makefile

```diff
@@ -45,6 +45,11 @@ MAGMA_DIR ?= ../magma
 CUDA_DIR  ?= $(or $(patsubst %/,%,$(dir $(patsubst %/,%,$(dir \
                $(shell which nvcc 2> /dev/null))))),/usr/local/cuda)
 
+# Check for PETSc in ../petsc
+ifneq ($(wildcard ../petsc/lib/libpetsc.*),)
```

Comment:
Is it ever installed like this?

Reply:
This works when `../petsc` is (a link to) the directory where PETSc was installed (at least when installed through spack).

Clearly, this logic can be improved. What is a good way to check for a local build in that location? For example, what's a good way to get `PETSC_ARCH` automatically from such local build location?

Reply:
@jedbrown, using the check as it is was convenient for me and it is consistent with how mfem, libxsmm, occa, and magma are setup up be auto-detected in `../<package-name>`. If you do not like it, feel free to modify it any way you like.

Reply:
But those packages have default builds (just run `make`) that leave the libraries and objects sitting next to the source. PETSc doesn't work that way so the user has to do something unnatural (create a special symlink; not a recommended method by PETSc) to use that code. Why didn't the old logic work for you?

Reply:
This section is all new - there was no default setting for `PETSC_DIR` before, so I always had to add it to the make command line or through the environment. Both of these approaches still work. The addition is to add auto-detection for PETSc for my setup which seems reasonable for other people to use too. However, as I already said, feel free to remove it if you do not like it, especially if you think it may break something.

Reply:
Would your environment find `pkg-config PETSc`? (I.e., installed to a system path or anything in `PKG_CONFIG_PATH`?) How does any other PETSc-using package in your environment find it? I want to be sure there is no ambiguity to the user about which PETSc will be found and also that users don't confuse build and install directories in trying to conform to a directory layout that may only exist on your machine.

Reply:
I do not setup `PKG_CONFIG_PATH` so `pkg-config PETSc` won't work for me right now.

When I need to use PETSc in MFEM, I set `PETSC_DIR` in my MFEM configuration file.